### PR TITLE
fix(Core/MovementHandler): fix crash at null guid in mover

### LINKED ISSUE #2409: Crash - Exploit hacker
Hello friends
I've been suffering attacks for 2 months

First attack was with macro / rum - Fixed
Second with program AccLeiTo-Crash - Fixed

Now I have a new problem with our log and I have no idea.
https://pastebin.com/bJidXUwC <<<<<
Operation System ***
Windows Server 2012 Server 4.0 (Version 6.2, Build 9200)

If anyone has an idea, it would help me a lot

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/83289840-crash-exploit-hacker?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2422: [Crash] [$100]
##### CURRENT BEHAVIOUR:
Server crashes for 2 days now, after the newest commit, with crash log.

##### STEPS TO REPRODUCE THE PROBLEM:
 I don't know. 

##### EXTRA NOTES:
 Crashlog:
https://gist.github.com/IceNet/d4aed5961be66e541dbbaa8afa96555b

##### BRANCH(ES):

master

##### AC HASH/COMMIT:
https://github.com/azerothcore/azerothcore-wotlk/commit/0fd565d5573dce91dc87874b6de2ca4b3f884bfa

##### OPERATING SYSTEM:
Windows Server 2016

##### MODULES:
1vs1
Account Mounts
Better Item Reloading
Duel Reset
Individual XP
PvP Titles
Teleporter
Transmog
Weekend XP

<bountysource-plugin>

---
Did you help close this issue? Go claim the **[$100 bounty](https://www.bountysource.com/issues/83501658-crash?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Moved condition for check bad guid

##### ISSUES ADDRESSED:
- Closes #2422
- Closes #2409 (maybe)

##### TESTS PERFORMED:
- Build, run, test in game (1 month) in Linux Debian 10

##### HOW TO TEST THE CHANGES:
- ...

##### KNOWN ISSUES AND TODO LIST:
- [x] Test in game

##### Target branch(es):
- [x] Master

### Copyright (C)
- Author fix @DraenorCore
- Testing @Wokwer

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Wokwer** [comment]
This code helps protect against hacker attacks. The bully has been terrorizing our server for several days, had to quarantine all the characters on level 1-5 and banned the creation of new ones so that the hacker could not enter the game world.
It was tested on my server when the hacker attacked him, on the one hand I was negotiating with the hooligan and on the other hand the programmers were writing the encoding, it was in real time, I was teasing the hacker, and he was attacking the server over and over again, for days, I was specifically increasing his anger so that he would apply all of his methods, and finally the programmers wrote the right code, the hacker became powerless, and it was possible to play on the server again. So I confirm 100% performance, without breakdowns in other parts of the code, because after that the players went through the dungeons several times, and did not face any problems, the game works very well.

**Comment by Helias** [comment]
I tested this PR too with `WPS`:
- with the attack using opcode `MSG_MOVE_HEARTBEAT` **the server still crashes**
- with the attack based on `MSG_MOVE_SET_FACING` the server does not crash :+1:

**Comment by DraenorCore** [comment]
Thanks. The fix works 👍

**Comment by IceNet** [comment]
Fix is working!
No crashes since monday 6pm

File: src/server/game/Handlers/MovementHandler.cpp [hunk 1]
```diff
@@ -319,22 +319,25 @@ void WorldSession::HandleMovementOpcodes(WorldPacket & recvData)
...
+    movementInfo.guid = guid;
+    ReadMovementInfo(recvData, &movementInfo);
 
     if (!movementInfo.pos.IsPositionValid())
     {
         recvData.rfinish();                     // prevent warnings spam
         return;
     }
 
+    recvData.rfinish();                         // prevent warnings spam
```
Comment (LichKing255) [inline src/server/game/Handlers/MovementHandler.cpp]: this is true ?

recvData It should finish when all bets have also been reviewed.
this line must move to between lines 504-505

Comment (FrancescoBorzi) [reply src/server/game/Handlers/MovementHandler.cpp]: @LichKing255 please open a PR

**Pull request merged**